Content.Server/Mining/MiningSystem.cs

```diff
@@ -15,38 +16,71 @@ public sealed class MiningSystem : EntitySystem
 {
     [Dependency] private readonly IPrototypeManager _proto = default!;
     [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly TagSystem _tagSystem = default!;
 
     /// <inheritdoc/>
     public override void Initialize()
     {
         base.Initialize();
-        SubscribeLocalEvent<OreVeinComponent, MapInitEvent>(OnMapInit);
         SubscribeLocalEvent<OreVeinComponent, DestructionEventArgs>(OnDestruction);
     }
 
     private void OnDestruction(EntityUid uid, OreVeinComponent component, DestructionEventArgs args)
     {
-        if (component.CurrentOre == null)
+        if (!_random.Prob(component.OreChance))
             return;
 
-        var proto = _proto.Index<OrePrototype>(component.CurrentOre);
+        string? currentOre = null;
 
-        if (proto.OreEntity == null)
-            return;
+        if (component.CurrentOre != null)
+        {
+            currentOre = component.CurrentOre;
+        }
+        else
+        {
+            string? weightedRandom = null;
 
-        var coords = Transform(uid).Coordinates;
-        var toSpawn = _random.Next(proto.MinOreYield, proto.MaxOreYield);
-        for (var i = 0; i < toSpawn; i++)
+            if (component.MappedTools != null)
+            {
+                var gatherer = args.Gatherer;
+
+                foreach (var (toolTag, mappedWeightedRandom) in component.MappedTools)
+                {
+                    if (gatherer != null && _tagSystem.HasTag(gatherer.Value, toolTag) || gatherer == null && toolTag == "Hand" || toolTag == "All")
+                    {
```

Comment:
Don't hardcode checks for hands here, just pass in gatheruser as I think I mentioned in the earlier review.

Reply:
please advise, I am able to check whether the user has hands, but I don't understand how to check if hands are being used without checking if the tool is null